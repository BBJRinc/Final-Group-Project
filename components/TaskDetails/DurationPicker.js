import React, { Component } from 'react';
import { View, Text, Picker, StyleSheet, StatusBar, Modal, TouchableHighlight } from 'react-native'
import { Header, Container } from 'native-base'
import IconE from 'react-native-vector-icons/Entypo'

export default class DurationPicker extends Component {
  constructor(props) {
    super(props)
    this.state = {
      hours: '00',
      minutes: '00',
    }
    this.updateHour = this.updateHour.bind(this)
    this.updateMin = this.updateMin.bind(this)

  }
  updateHour(value) {
    this.setState({ hours: value })
  }
  updateMin(value) {
    this.setState({ minutes: value })
  }

  render() {
    const { text, pickerView, headerBackground, pickerContainer, pickerViewLeft, pickerViewRight, picker, bottomView } = style
    return (
      <View>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.props.showDurationPicker}
        >
          <View style={{ backgroundColor: 'black' }}>
            <Header style={headerBackground}>
              <StatusBar backgroundColor='red' barStyle='light-content' />
              <Text style={text}>Duration</Text>
            </Header>
            <View style={ pickerContainer}>
              <View style={ pickerView }><Text style={ pickerViewLeft }>hours</Text><Text style={ pickerViewRight }>min</Text></View>
              <Picker selectedValue={this.state.hours} onValueChange={(value) => this.updateHour(value)} style={ picker }>
                <Picker.Item label="00" value="00" color='#fff' />
                <Picker.Item label="01" value="01" color='#fff' />
                <Picker.Item label="02" value="02" color='#fff' />
                <Picker.Item label="03" value="03" color='#fff' />
                <Picker.Item label="04" value="04" color='#fff' />
                <Picker.Item label="05" value="05" color='#fff' />
                <Picker.Item label="06" value="06" color='#fff' />
                <Picker.Item label="07" value="07" color='#fff' />
                <Picker.Item label="08" value="08" color='#fff' />
                <Picker.Item label="09" value="09" color='#fff' />
                <Picker.Item label="10" value="10" color='#fff' />
                <Picker.Item label="11" value="11" color='#fff' />
                <Picker.Item label="12" value="12" color='#fff' />
                <Picker.Item label="13" value="13" color='#fff' />
                <Picker.Item label="14" value="14" color='#fff' />
                <Picker.Item label="15" value="15" color='#fff' />
                <Picker.Item label="16" value="16" color='#fff' />
                <Picker.Item label="17" value="17" color='#fff' />
                <Picker.Item label="18" value="18" color='#fff' />
                <Picker.Item label="19" value="19" color='#fff' />
                <Picker.Item label="20" value="20" color='#fff' />
                <Picker.Item label="21" value="21" color='#fff' />
                <Picker.Item label="22" value="22" color='#fff' />
                <Picker.Item label="23" value="23" color='#fff' />
                <Picker.Item label="24" value="24" color='#fff' />
              </Picker>
              <Picker selectedValue={this.state.minutes} onValueChange={this.updateMin} style={{ width: 200, paddingRight: 50 }}>
                <Picker.Item label="00" value="00" color='#fff' />
                <Picker.Item label="15" value=".15" color='#fff' />
                <Picker.Item label="30" value=".30" color='#fff' />
                <Picker.Item label="45" value=".45" color='#fff' />
              </Picker>
            </View>
            <View style={ bottomView }>
                <Text style={text} onPress={(e)=>this.props.cancelDuration(e)}>Cancel</Text>
              <Text style={text} onPress={(e)=>this.props.saveDuration((e),this.state)}>Save</Text>
            </View>
          </View>
        </Modal>
      </View>
    )
  }
}

const style = {
  pickerView: {
    height: 35,
    borderWidth: .75,
    borderBottomColor: '#1b1b1b',
    borderTopColor: '#1b1b1b',
    width: 275,
    position: 'absolute',
    marginLeft: 50,
    paddingLeft: 70,
    marginTop: 90,
    justifyContent: 'space-between',
    flexDirection: 'row'
  },
  headerBackground:{
    backgroundColor: '#1b1b1b', 
    alignItems: 'center'
  }, 
  pickerContainer:{
    flexDirection: 'row', 
    backgroundColor: '#000', 
    justifyContent: 'center'
  },
  pickerViewLeft:{
    color: '#fff', 
    paddingTop: 8, 
    fontSize: 15
  },
  pickViewRight:{
    color: '#fff', 
    paddingTop: 8, 
    paddingRight: 15, 
    fontSize: 15
  },
  picker:{
    width: 200, 
    paddingLeft: 25
  },
  bottomView:{
    height: 35, 
    backgroundColor: '#1b1b1b', 
    justifyContent: 'space-between', 
    flexDirection: 'row', 
    alignItems: 'center', 
    paddingRight: 20, 
    paddingLeft: 20, 
    marginBottom: 300
  },
  text:{
    color:'#fff'
  }
}