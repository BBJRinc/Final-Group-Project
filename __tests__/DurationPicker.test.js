// Brandons tests

import React from 'react';
import DurationPicker from '../components/TaskDetails/DurationPicker';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<DurationPicker />).toJSON();
    expect(tree).toMatchSnapshot();
});

