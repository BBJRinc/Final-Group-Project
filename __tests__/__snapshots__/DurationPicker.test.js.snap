// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "black",
        }
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F8F8F8",
                "borderBottomColor": "#a7a6ab",
                "borderBottomWidth": 0.5,
                "elevation": 3,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "left": 0,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 15,
                "right": 0,
                "top": 0,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#1b1b1b",
              },
            ]
          }
          virtual={undefined}
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#000",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#1b1b1b",
              "borderTopColor": "#1b1b1b",
              "borderWidth": 0.75,
              "flexDirection": "row",
              "height": 35,
              "justifyContent": "space-between",
              "marginLeft": 50,
              "marginTop": 90,
              "paddingLeft": 70,
              "position": "absolute",
              "width": 275,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 15,
                "paddingTop": 8,
              }
            }
          >
            hours
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          >
            min
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 25,
              "width": 200,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "00",
                  "textColor": 4294967295,
                  "value": "00",
                },
                Object {
                  "label": "01",
                  "textColor": 4294967295,
                  "value": "01",
                },
                Object {
                  "label": "02",
                  "textColor": 4294967295,
                  "value": "02",
                },
                Object {
                  "label": "03",
                  "textColor": 4294967295,
                  "value": "03",
                },
                Object {
                  "label": "04",
                  "textColor": 4294967295,
                  "value": "04",
                },
                Object {
                  "label": "05",
                  "textColor": 4294967295,
                  "value": "05",
                },
                Object {
                  "label": "06",
                  "textColor": 4294967295,
                  "value": "06",
                },
                Object {
                  "label": "07",
                  "textColor": 4294967295,
                  "value": "07",
                },
                Object {
                  "label": "08",
                  "textColor": 4294967295,
                  "value": "08",
                },
                Object {
                  "label": "09",
                  "textColor": 4294967295,
                  "value": "09",
                },
                Object {
                  "label": "10",
                  "textColor": 4294967295,
                  "value": "10",
                },
                Object {
                  "label": "11",
                  "textColor": 4294967295,
                  "value": "11",
                },
                Object {
                  "label": "12",
                  "textColor": 4294967295,
                  "value": "12",
                },
                Object {
                  "label": "13",
                  "textColor": 4294967295,
                  "value": "13",
                },
                Object {
                  "label": "14",
                  "textColor": 4294967295,
                  "value": "14",
                },
                Object {
                  "label": "15",
                  "textColor": 4294967295,
                  "value": "15",
                },
                Object {
                  "label": "16",
                  "textColor": 4294967295,
                  "value": "16",
                },
                Object {
                  "label": "17",
                  "textColor": 4294967295,
                  "value": "17",
                },
                Object {
                  "label": "18",
                  "textColor": 4294967295,
                  "value": "18",
                },
                Object {
                  "label": "19",
                  "textColor": 4294967295,
                  "value": "19",
                },
                Object {
                  "label": "20",
                  "textColor": 4294967295,
                  "value": "20",
                },
                Object {
                  "label": "21",
                  "textColor": 4294967295,
                  "value": "21",
                },
                Object {
                  "label": "22",
                  "textColor": 4294967295,
                  "value": "22",
                },
                Object {
                  "label": "23",
                  "textColor": 4294967295,
                  "value": "23",
                },
                Object {
                  "label": "24",
                  "textColor": 4294967295,
                  "value": "24",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "paddingRight": 50,
              "width": 200,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "00",
                  "textColor": 4294967295,
                  "value": "00",
                },
                Object {
                  "label": "15",
                  "textColor": 4294967295,
                  "value": ".15",
                },
                Object {
                  "label": "30",
                  "textColor": 4294967295,
                  "value": ".30",
                },
                Object {
                  "label": "45",
                  "textColor": 4294967295,
                  "value": ".45",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1b1b1b",
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-between",
            "marginBottom": 300,
            "paddingLeft": 20,
            "paddingRight": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Cancel
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
